### PR TITLE
Store press_counter register values as soon as possible

### PR DESCRIPTION
**Pull request by KraPete**
## Summary by CodeRabbit

- **New Features**
	- Immediate storage of `press_counter` register values to prevent outdated publications after restarts.
	- Introduction of a new `press_counter` register type in device templates, avoiding the first zero value publication after a reset.

- **Bug Fixes**
	- Removal of zero press counter value publications after device resets, resolving false positives in `whenChanged` conditions.

- **Improvements**
	- Enhanced error handling and logging for better monitoring and debugging of device communications.
	- Updated documentation to clarify new features and functionalities related to device communication parameters and press counters.
	- Refined input handling in device configurations by changing `press_counter` to `input` for various channels.

### CODE REVIEW

File: templates/config-wb-mrm2-mini.json.jinja [hunk 1]
```diff
@@ -573,7 +573,7 @@
             },
             {
                 "name": "Input {{in_num}} counter",
-                "reg_type": "press_counter",
+                "reg_type": "input",
```
[the remark this answers is not in the document]
Comment (KraPete) [reply templates/config-wb-mrm2-mini.json.jinja:576]: useless

File: templates/config-wb-mao4-20ma.json.jinja [hunk 2]
```diff
@@ -516,7 +516,7 @@
             {
                 "id": "input_{{index + 1}}_counter",
                 "name": "Input {{index + 1}} Counter",
-                "reg_type": "press_counter",
+                "reg_type": "input",
```
[the remark this answers is not in the document]
Comment (KraPete) [reply templates/config-wb-mao4-20ma.json.jinja:519]: useless

File: templates/config-wb-mwac.json.jinja [hunk 3]
```diff
@@ -333,7 +333,7 @@
             },
             {
                 "name": "{{ letter }}{{ in_num - in_num_offset }} Counter",
-                "reg_type": "press_counter",
+                "reg_type": "input",
```
[the remark this answers is not in the document]
Comment (KraPete) [reply templates/config-wb-mwac.json.jinja:336]: useless

File: README.md [hunk 4]
```diff
@@ -1079,6 +1081,16 @@ It's designed to be used on [Wiren Board](https://wirenboard.com/en/) family of
 ```
 Ответы с ошибками аналогичны прямому чтению и записи в порт.
 
+### Обработка счётчиков нажатий
+
+В Modbus устройствах Wiren Board есть регистры, которые показывают число нажатий (простых, длинных, двойных и т.п.). Рекомендуется использовать их вместо прямого чтения состояния входов для организации реакции на нажатия во внешних системах. Обычно это 16-битные регистры, значение которых монотонно увеличивается. При достижении максимального значения отсчёт начинается заново с 0. При перезагрузке устройства значение сбрасывается в 0.
```
Comment (aadegtyarev) [inline README.md:1086]: >показывают число нажатий (простых, длинных, двойных и т.п.).

«простых» — не используем этот термин, там так:
одиночных, двойных, длинных

File: README.md [hunk 5]
```diff
@@ -1079,6 +1081,16 @@ It's designed to be used on [Wiren Board](https://wirenboard.com/en/) family of
...
+
+Для описания таких регистров в шаблонах используется тип (`reg_type`) `press_counter`.
+
+Сервис отслеживает перезагрузку устройства и не публикует 0, если он получен сразу после перезагрузки. Это сделано для того, чтобы избежать ложных срабатываний при перезагрузке устройства. Для корректной работы этой функции прошивка устройства должна поддерживать [события "Быстрого Модбаса"](https://wirenboard.com/wiki/Fast_Modbus#%D0%A1%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D1%8F_%D0%BD%D0%B0_%D1%88%D0%B8%D0%BD%D0%B5).
+
+Сервис записывает состояния регистров во внутреннюю базу и при старте публикует эти сохранённые значения. Сохранённые значения сбрасываются на диск раз в минуту. Если значения не успели записать на диск, при запуске сервиса будет опубликовано какое-то старое значение, а потом текущее, полученное чтением из регистра. При записи значений регистров с типом `press_counter` база сразу сбрасывается на диск, чтобы избежать публикации неактуальных значений.
```
Comment (aadegtyarev) [inline README.md:1092]: >Сервис записывает состояния регистров 

Обычно сервис записывает…

File: README.md (hunk 4 above)
Comment (KraPete) [reply README.md]: Поправил

File: src/serial_port_driver.cpp [hunk 6]
```diff
@@ -259,6 +259,13 @@ TControlArgs TSerialPortDriver::From(const PDeviceChannel& channel)
         args.SetEnumValueTitles(it.first, it.second);
     }
 
+    if (channel->Registers.cend() != std::find_if(channel->Registers.cbegin(), channel->Registers.cend(), [](auto reg) {
+            return reg->TypeName == "press_counter";
+        }))
```
Comment (sikmir) [inline src/serial_port_driver.cpp:264]:
```suggestion
    if (std::any_of(channel->Registers.cbegin(), channel->Registers.cend(), [](const auto& reg) {
            return reg->TypeName == "press_counter";
        }))
```

File: README.md (hunk 5 above)
Comment (KraPete) [reply README.md]: Поправил

**Review by sikmir — APPROVED** [review]

**Review by aadegtyarev — APPROVED** [review]

**Pull request merged**